sequence_processing_pipeline/NuQCJob.py

```diff
@@ -0,0 +1,368 @@
+
+        """
+        super().__init__(fastq_root_dir,
+                         output_path,
+                         'QCJob',
+                         [fastp_path, minimap2_path, samtools_path],
+                         max_array_length,
+                         modules_to_load=modules_to_load)
+        self.sample_sheet_path = sample_sheet_path
+        self._file_check(self.sample_sheet_path)
+        metadata = self._process_sample_sheet()
+        self.sample_ids = metadata['sample_ids']
+        self.project_data = metadata['projects']
+        self.needs_trimming = metadata['needs_adapter_trimming']
+        self.nprocs = 16 if nprocs > 16 else nprocs
+        self.chemistry = metadata['chemistry']
+        # instead of a list of individual .mmi files,  minimap_database_paths
+        # is now a path to a directory of .mmi files.
+        self.minimap_database_paths = minimap_database_paths
+        self.queue_name = queue_name
+        self.node_count = node_count
+        self.wall_time_limit = wall_time_limit
+        self.jmem = jmem
+        self.fastp_path = fastp_path
+        self.minimap2_path = minimap2_path
+        self.samtools_path = samtools_path
+        self.qiita_job_id = qiita_job_id
+        self.pool_size = pool_size
+        self.suffix = 'fastq.gz'
+
+        ### BEGIN NEW PARAMS
+        # sizes in GB
+        self.max_file_list_size_in_gb = 2  # max size of GB data to process in a job
+        self.data_location = "/panfs/dtmcdonald/human-depletion/t2t-only/*/*"
+
+        datestamp = date.today().strftime("%Y.%m.%d")
+
+        self.batch_prefix = join(self.output_path,
+                                 f"1-hd-split-pangenome-pe-{datestamp}")
+
+        self.output = "/panfs/dtmcdonald/human-depletion/pangenome-adapter-filter"
+        self.tmpdir = "/scratch/tcga_compute/tmp"
+        ### END NEW PARAMS
+
+        self.minimum_bytes = 3100
+
+        if not isinstance(self.needs_trimming, bool):
+            raise ValueError("needs_adapter_trimming must be boolean.")
+
+        # Validate project settings in [Bioinformatics]
+        for project in self.project_data:
+            if project['ForwardAdapter'] == 'NA':
+                project['ForwardAdapter'] = None
+
+            if project['ReverseAdapter'] == 'NA':
+                project['ReverseAdapter'] = None
+
+            if project['ForwardAdapter'] is None:
+                if project['ReverseAdapter'] is not None:
+                    raise ValueError(("ForwardAdapter is declared but not "
+                                      "ReverseAdapter."))
+
+            if project['ReverseAdapter'] is None:
+                if project['ForwardAdapter'] is not None:
+                    raise ValueError(("ReverseAdapter is declared but not "
+                                      "ForwardAdapter."))
+
+            if not isinstance(project['HumanFiltering'], bool):
+                raise ValueError("needs_adapter_trimming must be boolean.")
```

Comment:
the adapter stuff doesn't seem to be used downstream but I maybe missing something. Our current strategy is to filter for all known adapters (relative to fastp's known set). If the adapters used bu the wet lab _are not_ included in that fasta file, then we should add them rather.

Reply:
Lines moved into a validation function, _validate_project_data(). We will confirm that the adapters used by the wet-lab and minimap2 are part of the fasta file.